mcs/class/System.Windows.Forms/System.Windows.Forms/ListViewGroup.cs

```diff
@@ -84,15 +84,23 @@ private ListViewGroup(SerializationInfo info, StreamingContext context)
 			header_alignment = (HorizontalAlignment)info.GetInt32("HeaderAlignment");
 			tag = info.GetValue("Tag", typeof(object));
 
-			int count = info.GetInt32("ListViewItemCount");
-			if (count > 0) {
-				if(items == null)
-				items = new ListView.ListViewItemCollection(list_view_owner);
-
-				for (int i = 0; i < count; i++)
-				{
-					items.Add((ListViewItem)info.GetValue(string.Format("ListViewItem_{0}", i), typeof(ListViewItem)));
-				}
+			int count = 0;
+			try {
+				// Mono backwards compat
+        			count = info.GetInt32("ListViewItemCount");
+			} catch (SerializationException e) {
+			}
+			try {
+        			count = info.GetInt32("ItemsCount");
```

Comment:
This will always override the previous value unless an exception is thrown. It should probably be in the `catch` block above to avoid unnecessary exception penalty.

Reply:
...and if you are going for some serialization compatibility between dotnet/winforms and Mono then it would probably make sense to update the serialization code too and change the preferred order here.

Reply:
Thanks for the review! Sorry I didn't reply earlier, I wasn't home for some days.

I totally forgot about the serialization case as it didn't trigger any problems in my use-case, but I'm back this weekend and will update the PR.